### PR TITLE
Search for lib in `java.library.path` instead of hardcoded path

### LINKED ISSUE #72: Use native system PATH to search for fuse-lib
To load the fuse-library, we keep a list of possible locations for linux:
https://github.com/cryptomator/fuse-nio-adapter/blob/117771993be8cd3e813acc08c34ec61cad61a0a8/src/main/java/org/cryptomator/frontend/fuse/mount/LinuxFuseMountProvider.java#L42-L48

But since there is a vast majority of Linux systems, we should consider discarding the list and let the native lib loader just scrape the system PATH.

### PR DESCRIPTION
**Pull request by nicomem**
Resolve #72

### CODE REVIEW

**Comment by overheadhunter** [comment]
Both `System.mapLibraryName` and looking in `java.library.path` is what Java does per default (via `System.loadLibrary`), however what happens here is that it simply adds `"lib" + name + ".so"` on Linux, which fails if the file name is e.g. Debian's `libfuse3.so.3`.

Hence we had to ditch this default approach. 

What we discussed internally: Adding a separate system property, however some Cryptomator distributions (like the AppImage) don't allow to manually override those. Our best option currently is to look in some well-known locations and fallback to `System.loadLibrary` and hope that there is no additional suffix after `.so`.

**Comment by infeo** [comment]
> fallback to System.loadLibrary and hope that there is no additional suffix after .so.

Some systems create a symlink to the newest version of a library, some not. If we add this last ditch attempt, we need to ensure to use the correct fuse version (by calling [fuse_version()](http://libfuse.github.io/doxygen/fuse__common_8h.html#ae04b2f721003f5540c71be65396c53c8))

**Comment by overheadhunter** [comment]
> > fallback to System.loadLibrary and hope that there is no additional suffix after .so.
> 
> Some systems create a symlink to the newest version of a library, some not. If we add this last ditch attempt, we need to ensure to use the correct fuse version (by calling [fuse_version()](http://libfuse.github.io/doxygen/fuse__common_8h.html#ae04b2f721003f5540c71be65396c53c8))

The major version (which ensures API-compatibility) is part of the lib name (`fuse3` instead of `fuse`)

**Review by overheadhunter — CHANGES_REQUESTED** [review]
I just opened https://github.com/cryptomator/jfuse/pull/37 to fallback to the `System.loadLibrary` mechanism.

fuse-nio-adapter should not duplicate what Java is already doing. Instead `builder.setLibraryPath(...)` is intended to be used to attempt loading from a _verified_ path to an existing .so file.

If no such file could be found in those places, **do not set** the library path, which will then cause jfuse to fallback to `System.loadLibrary(...)`.

In other words: After updating jfuse, we need something like this:

```java
Arrays.stream(LIB_PATHS).map(Path::of).filter(Files::exists).map(Path::toString).findAny().ifPresent(builder::setLibraryPath);
```

**Comment by nicomem** [comment]
Since there will now be a fallback mecanism in case no element in `LIB_PATHS` is present, the `isSupported()` method will also need to remove the first test and simply become `return isFusermount3Installed();`, right?

**Comment by overheadhunter** [comment]
> Since there will now be a fallback mecanism in case no element in `LIB_PATHS` is present, the `isSupported()` method will also need to remove the first test and simply become `return isFusermount3Installed();`, right?

Right!

Furthermore we need to remove `/ust/lib/libfuse3.so` from our array, as this is now handled via the fallback mechanism (assuming `/usr/lib` is in `java.library.path` on any sane system). Otherwise it would take precedence over e.g. `/usr/lib64`, which is what causes cryptomator/cryptomator#3127.

**Comment by overheadhunter** [comment]
Also, bump jfuse to 0.6.2

**Comment by infeo** [comment]
the latter.
> 2. or base our next Cryptomator release on JDK 21? (see https://github.com/cryptomator/cryptomator/pull/3129)

**Review by overheadhunter — APPROVED** [review]
